test/python/transpiler/test_high_level_synthesis.py

```diff
@@ -2607,5 +2607,57 @@ def test_annotated_mcx(self):
         self.assertEqual(Operator(qc), Operator(qct))
 
 
+@ddt
+class TestPauliEvolutionSynthesisPlugins(QiskitTestCase):
+    """Tests related to plugins for PauliEvolutionGate."""
+
+    def test_supported_names(self):
+        """Test that "default" and "rustiq" plugins do exist."""
+        supported_plugin_names = high_level_synthesis_plugin_names("PauliEvolution")
+        self.assertIn("default", supported_plugin_names)
+        self.assertIn("rustiq", supported_plugin_names)
+
+    @data("default", "rustiq")
+    def test_correctness(self, plugin_name):
```

Comment:
To check correctness we'd have to compare `qct` to the matrix-based evolution of the Trotter steps, see https://github.com/Qiskit/qiskit/blob/2efb59ce4cd3c2d8721390648cdca6d2fd41f792/test/python/circuit/library/test_evolution_gate.py#L42. But since that's already tested for the default decomposition, we could just check whether the `rustiq` plugin equals the `default`.

Reply:
I believe that this is precisely what gets done internally when calculating `Operator(qc)`.